Docs/Documentation/SimpleRbacAuthorize.md

```diff
@@ -52,49 +52,90 @@ The ```default_role``` will be used to set the role of the registered users by d
+    * If `expected_value` is **not** a callable/object then it's simply casted to boolean
+    * The `*` is checked and if found the result is inverted
+    * The final boolean value is **the result of permission** checker. This means if it is `false` then no other permissions are checked and the user is denied access.
+    For this reason the `allowed` key must be placed at the end of permission since no other rules are executed after it
 
-Permission Callbacks
------------------
-You could use a callback in your 'allowed' to process complex authentication, like
+**Notes**:
+
+* For Superadmin access (permission to access ALL THE THINGS in your app) there is a specific Authorize Object provided
+* Permissions that do not have `controller` and/or `action` keys (or the inverted versions) are automatically discarded in order to prevent errors.
+If you need to match all controllers/actions you can explicitly do `'contoller' => '*'`
+* Key `user` (or the inverted version) is illegal (as it's impossible to match an array) and any permission containing it will be discarded
+* If the permission is discarded for the reasons stated above, a debug message will be logged
+
+**Permission Callbacks**: you could use a callback in your 'allowed' to process complex authentication, like
   - ownership
   - permissions stored in your database
   - permission based on an external service API call
 
 Example *ownership* callback, to allow users to edit their own Posts:
 
 ```php
-    'allowed' => function (array $user, $role, Request $request) {
-        $postId = Hash::get($request->params, 'pass.0');
-        $post = TableRegistry::get('Posts')->get($postId);
-        $userId = Hash::get($user, 'id');
+    'allowed' => function (array $user, $role, \Cake\Network\Request $request) {
+        $postId = \Cake\Utility\Hash::get($request->params, 'pass.0');
+        $post = \Cake\ORM\TableRegistry::get('Posts')->get($postId);
+        $userId = $user['id'];
         if (!empty($post->user_id) && !empty($userId)) {
             return $post->user_id === $userId;
         }
         return false;
     }
 ```
 
-Permission Rules
-----------------
-You could use an instance of the \CakeDC\Users\Auth\Rules\Rule interface to reuse your custom rules
-Examples:
+**Permission Rules**: If you see that you are duplicating logic in your callables, you can create rule class to re-use the logic.
+For example, the above ownership callback is included in CakeDC\Users as `Owner` rule
 ```php
-'allowed' => new Owner() //will pick by default the post id from the first pass param
+'allowed' => new \CakeDC\Users\Auth\Rules\Owner() //will pick by default the post id from the first pass param
 ```
 Check the [Owner Rule](OwnerRule.md) documentation for more details
 
+Creating rule classes
+---------------------
+
+The only requirement is to implement `\CakeDC\Users\Auth\Rules\Rule` interface which has one method:
```

Comment:
I don't see mentions to the Interface convention here http://book.cakephp.org/3.0/en/intro/conventions.html or http://www.php-fig.org/psr/psr-2/ but makes sense for me to add it and match the interfaces defined in the core. Let's do this BBC change in a future milestone.